post_processing/compare_outputs.jl

```diff
@@ -63,12 +65,15 @@ compare(a::F, b::F, pn0 = "") where {F <: Fields.Field} =
     compare(true, a, b, pn0)
 function compare(zero_error, a::F, b::F, pn0 = "") where {F <: Fields.Field}
     isempty(propertynames(a)) || return _compare(zero_error, a, b, pn0)
-    err = abs.(Array(parent(a)) .- Array(parent(b)))
-    if !(maximum(err) == 0.0)
+    abs_err = abs.(Array(parent(a)) .- Array(parent(b)))
+    rel_err =
+        norm(Array(parent(a)) .- Array(parent(b))) ./ norm(Array(parent(a)))
+    if !(maximum(abs_err) == 0.0) || !(maximum(rel_err) == 0.0)
         @warn "--- Failed field comparison $pn0"
         @show a
         @show b
-        @show maximum(err)
+        @show maximum(abs_err)
+        @show rel_err
```

Comment:
rather than using `@show`, it would be better to  make these arguments to `@warn`

Reply:
Do you suggest to use multiple `@warn` lines? Like: 
```suggestion
        @warn "First field: $a"
        @warn "Second field: $b"
        @warn "Max absolute error $maximum(abs_err)"
        @warn "Relative error $rel_err"
```
? 
That would be a bunch of warnings

Reply:
no, you can pass args to `@warn` (like all logging macros). They can be either `key=arg`, or simply `arg` (where the key will be the name of the arg)
```
     @warn "Failed field comparison $pn0" a b max_abs_err=maximum(abs_err) rel_arr
```

Reply:
See https://docs.julialang.org/en/v1/stdlib/Logging/#Logging.@logmsg